lrpar/src/lib/ctbuilder.rs

```diff
@@ -92,6 +92,59 @@ where
 {
 }
 
+/// The scope of public visibility.
+#[derive(Clone,PartialEq,Eq,Debug,Hash)]
+pub enum Scope {
+    /// Represents `pub(crate)`
+    Crate,
+    /// Represents `pub(self)`
+    _Self,
+    /// Represents `pub(super)`
+    Super,
+    /// `pub(in String)`
+    In(String),
+}
+
+/// Visibility controls how public generated code is.
+#[derive(Clone,PartialEq,Eq,Debug,Hash)]
+pub enum Visibility {
+    Private,
+    Public(Option<Scope>),
```

Comment:
I wonder if we're being a bit too clever here? I think it's fine for us to have a single `enum` with `Public/Private/Crate/Self_/In(String)` which would be easier to use. That would also get rid of the `Option` and `Scope`.

Reply:
Sure.